File: packages/i18nify-go/modules/geo/format_address.go
package geo

import (
	"fmt"
	"strings"
)

// FormatAddress formats address components using a caller-supplied template string.
func FormatAddress(template string, components AddressComponents) (string, error) {
	if strings.TrimSpace(template) == "" {
		return "", fmt.Errorf("template must be a non-empty string")
	}

	replacer := strings.NewReplacer(
		"{name}", components.Name,
		"{organization}", components.Organization,
		"{street_address}", components.StreetAddress,
		"{city}", components.City,
		"{state}", components.State,
		"{zip}", components.Zip,
		"{district}", components.District,
		"{sorting_code}", components.SortingCode,
	)
	substituted := replacer.Replace(template)

	rawLines := strings.Split(substituted, "\n")
	formatted := make([]string, 0, len(rawLines))
	for _, line := range rawLines {
		if trimmed := strings.TrimSpace(line); trimmed != "" {
			formatted = append(formatted, trimmed)
		}
	}

	return strings.Join(formatted, "\n"), nil
}

File: packages/i18nify-js/src/modules/geo/formatAddress.ts
import { withErrorBoundary } from '../../common/errorBoundary';
import { AddressComponents } from './types';

const formatAddress = (
  template: string,
  components: AddressComponents,
): string => {
  if (!template || typeof template !== 'string' || !template.trim())
    throw new Error('template must be a non-empty string.');

  const substituted = template
    .replace(/{name}/g, components.name ?? '')
    .replace(/{organization}/g, components.organization ?? '')
    .replace(/{street_address}/g, components.street_address ?? '')
    .replace(/{city}/g, components.city ?? '')
    .replace(/{state}/g, components.state ?? '')
    .replace(/{zip}/g, components.zip ?? '')
    .replace(/{district}/g, components.district ?? '')
    .replace(/{sorting_code}/g, components.sorting_code ?? '');

  return substituted
    .split('\n')
    .map((line) => line.trim())
    .filter((line) => line.length > 0)
    .join('\n');
};

export default withErrorBoundary<typeof formatAddress>(formatAddress);
